File: src/main/java/com/rdvdev2/timetravelmod/impl/ModItems.java
package com.rdvdev2.timetravelmod.impl;

import com.rdvdev2.timetravelmod.impl.common.item.CreativeTimeMachineItem;
import com.rdvdev2.timetravelmod.impl.common.item.EngineerBookItem;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.block.Block;
import net.minecraft.item.BlockItem;
import net.minecraft.item.BucketItem;
import net.minecraft.item.Item;
import net.minecraft.item.Items;
import net.minecraft.util.registry.Registry;

public class ModItems {

    public static final Item TIME_CRYSTAL = new Item(new FabricItemSettings().group(Mod.ITEM_GROUP));
    public static final Item CONTROLLER_CIRCUIT = new Item(new FabricItemSettings().group(Mod.ITEM_GROUP));
    public static final Item HEAVY_INGOT = new Item(new FabricItemSettings().group(Mod.ITEM_GROUP));
    public static final Item CREATIVE_TIME_MACHINE = new CreativeTimeMachineItem(new FabricItemSettings().group(Mod.ITEM_GROUP).maxCount(1));
    public static final Item ENGINEER_BOOK = new EngineerBookItem(new FabricItemSettings().group(Mod.ITEM_GROUP).maxCount(1));
    public static final Item COMMUNICATIONS_CIRCUIT = new Item(new FabricItemSettings().group(Mod.ITEM_GROUP));

    public static final Item TIME_CRYSTAL_ORE = createBlockItem(ModBlocks.TIME_CRYSTAL_ORE);
    public static final Item TIME_MACHINE_BASIC_BLOCK = createBlockItem(ModBlocks.TIME_MACHINE_BASIC_BLOCK);
    public static final Item TIME_MACHINE_CABLE_PANEL = createBlockItem(ModBlocks.TIME_MACHINE_CABLE_PANEL);
    public static final Item TIME_MACHINE_PILLAR = createBlockItem(ModBlocks.TIME_MACHINE_PILLAR);
    public static final Item TIME_MACHINE_HORIZONTAL_VENTILATION = createBlockItem(ModBlocks.TIME_MACHINE_HORIZONTAL_VENTILATION);
    public static final Item TIME_MACHINE_VERTICAL_VENTILATION = createBlockItem(ModBlocks.TIME_MACHINE_VERTICAL_VENTILATION);
    public static final Item TIME_MACHINE_CORE = createBlockItem(ModBlocks.TIME_MACHINE_CORE);
    public static final Item TIME_MACHINE_CONTROL_PANEL = createBlockItem(ModBlocks.TIME_MACHINE_CONTROL_PANEL);
    public static final Item HEAVY_BLOCK = createBlockItem(ModBlocks.HEAVY_BLOCK);
    public static final Item REINFORCED_HEAVY_BLOCK = createBlockItem(ModBlocks.REINFORCED_HEAVY_BLOCK);
    public static final Item ANOMALOUS_ATEMPORAL_VOID = createBlockItem(ModBlocks.ANOMALOUS_ATEMPORAL_VOID);
    public static final Item TEMPORAL_CAULDRON = createBlockItem(ModBlocks.TEMPORAL_CAULDRON);
    public static final Item GUNPOWDER_WIRE = createBlockItem(ModBlocks.GUNPOWDER_WIRE);
    public static final Item TIME_MACHINE_TRACKER = createBlockItem(ModBlocks.TIME_MACHINE_TRACKER);
    public static final Item TIME_MACHINE_RECALLER = createBlockItem(ModBlocks.TIME_MACHINE_RECALLER);

    private static Item createBlockItem(Block block) {
        return new BlockItem(block, new FabricItemSettings().group(Mod.ITEM_GROUP));
    }

    public static void register() {
        registerItem("time_crystal", TIME_CRYSTAL);
        registerItem("controller_circuit", CONTROLLER_CIRCUIT);
        registerItem("heavy_ingot", HEAVY_INGOT);
        registerItem("creative_time_machine", CREATIVE_TIME_MACHINE);
        registerItem("engineer_book", ENGINEER_BOOK);
        registerItem("communications_circuit", COMMUNICATIONS_CIRCUIT);

        registerItem("time_crystal_ore", TIME_CRYSTAL_ORE);
        registerItem("time_machine_basic_block", TIME_MACHINE_BASIC_BLOCK);
        registerItem("time_machine_cable_panel", TIME_MACHINE_CABLE_PANEL);
        registerItem("time_machine_pillar", TIME_MACHINE_PILLAR);
        registerItem("time_machine_horizontal_ventilation", TIME_MACHINE_HORIZONTAL_VENTILATION);
        registerItem("time_machine_vertical_ventilation", TIME_MACHINE_VERTICAL_VENTILATION);
        registerItem("time_machine_control_panel", TIME_MACHINE_CONTROL_PANEL);
        registerItem("time_machine_core", TIME_MACHINE_CORE);
        registerItem("heavy_block", HEAVY_BLOCK);
        registerItem("reinforced_heavy_block", REINFORCED_HEAVY_BLOCK);
        registerItem("anomalous_atemporal_void", ANOMALOUS_ATEMPORAL_VOID);
        registerItem("temporal_cauldron", TEMPORAL_CAULDRON);
        registerItem("gunpowder_wire", GUNPOWDER_WIRE);
        registerItem("time_machine_tracker", TIME_MACHINE_TRACKER);
        registerItem("time_machine_recaller", TIME_MACHINE_RECALLER);
    }

    private static void registerItem(String path, Item item) {
        Registry.register(Registry.ITEM, Mod.identifier(path), item);
    }
}

File: src/main/kotlin/com/rdvdev2/timetravelmod/impl/ModItems.kt
package com.rdvdev2.timetravelmod.impl

import com.rdvdev2.timetravelmod.impl.Mod.identifier
import com.rdvdev2.timetravelmod.impl.common.item.CreativeTimeMachineItem
import com.rdvdev2.timetravelmod.impl.common.item.EngineerBookItem
import net.fabricmc.fabric.api.item.v1.FabricItemSettings
import net.minecraft.block.Block
import net.minecraft.item.BlockItem
import net.minecraft.item.Item
import net.minecraft.util.registry.Registry

object ModItems {

    @JvmField
    val TIME_CRYSTAL = Item(FabricItemSettings().group(Mod.ITEM_GROUP))
    val CONTROLLER_CIRCUIT = Item(FabricItemSettings().group(Mod.ITEM_GROUP))
    val HEAVY_INGOT = Item(FabricItemSettings().group(Mod.ITEM_GROUP))
    @JvmField
    val CREATIVE_TIME_MACHINE: Item = CreativeTimeMachineItem(FabricItemSettings().group(Mod.ITEM_GROUP).maxCount(1))
    val ENGINEER_BOOK: Item = EngineerBookItem(FabricItemSettings().group(Mod.ITEM_GROUP).maxCount(1))
    val COMMUNICATIONS_CIRCUIT = Item(FabricItemSettings().group(Mod.ITEM_GROUP))
    val TIME_CRYSTAL_ORE = ModBlocks.TIME_CRYSTAL_ORE.createBlockItem()
    val TIME_MACHINE_BASIC_BLOCK = ModBlocks.TIME_MACHINE_BASIC_BLOCK.createBlockItem()
    val TIME_MACHINE_CABLE_PANEL = ModBlocks.TIME_MACHINE_CABLE_PANEL.createBlockItem()
    val TIME_MACHINE_PILLAR = ModBlocks.TIME_MACHINE_PILLAR.createBlockItem()
    val TIME_MACHINE_HORIZONTAL_VENTILATION = ModBlocks.TIME_MACHINE_HORIZONTAL_VENTILATION.createBlockItem()
    val TIME_MACHINE_VERTICAL_VENTILATION = ModBlocks.TIME_MACHINE_VERTICAL_VENTILATION.createBlockItem()
    val TIME_MACHINE_CORE = ModBlocks.TIME_MACHINE_CORE.createBlockItem()
    val TIME_MACHINE_CONTROL_PANEL = ModBlocks.TIME_MACHINE_CONTROL_PANEL.createBlockItem()
    val HEAVY_BLOCK = ModBlocks.HEAVY_BLOCK.createBlockItem()
    @JvmField
    val REINFORCED_HEAVY_BLOCK = ModBlocks.REINFORCED_HEAVY_BLOCK.createBlockItem()
    val ANOMALOUS_ATEMPORAL_VOID = ModBlocks.ANOMALOUS_ATEMPORAL_VOID.createBlockItem()
    val TEMPORAL_CAULDRON = ModBlocks.TEMPORAL_CAULDRON.createBlockItem()
    val GUNPOWDER_WIRE = ModBlocks.GUNPOWDER_WIRE.createBlockItem()
    @JvmField
    val TIME_MACHINE_TRACKER = ModBlocks.TIME_MACHINE_TRACKER.createBlockItem()
    val TIME_MACHINE_RECALLER = ModBlocks.TIME_MACHINE_RECALLER.createBlockItem()

    fun register() {
        TIME_CRYSTAL.registerAs("time_crystal")
        CONTROLLER_CIRCUIT.registerAs("controller_circuit")
        HEAVY_INGOT.registerAs("heavy_ingot")
        CREATIVE_TIME_MACHINE.registerAs("creative_time_machine")
        ENGINEER_BOOK.registerAs("engineer_book")
        COMMUNICATIONS_CIRCUIT.registerAs("communications_circuit")
        TIME_CRYSTAL_ORE.registerAs("time_crystal_ore")
        TIME_MACHINE_BASIC_BLOCK.registerAs("time_machine_basic_block")
        TIME_MACHINE_CABLE_PANEL.registerAs("time_machine_cable_panel")
        TIME_MACHINE_PILLAR.registerAs("time_machine_pillar")
        TIME_MACHINE_HORIZONTAL_VENTILATION.registerAs("time_machine_horizontal_ventilation")
        TIME_MACHINE_VERTICAL_VENTILATION.registerAs("time_machine_vertical_ventilation")
        TIME_MACHINE_CONTROL_PANEL.registerAs("time_machine_control_panel")
        TIME_MACHINE_CORE.registerAs("time_machine_core")
        HEAVY_BLOCK.registerAs("heavy_block")
        REINFORCED_HEAVY_BLOCK.registerAs("reinforced_heavy_block")
        ANOMALOUS_ATEMPORAL_VOID.registerAs("anomalous_atemporal_void")
        TEMPORAL_CAULDRON.registerAs("temporal_cauldron")
        GUNPOWDER_WIRE.registerAs("gunpowder_wire")
        TIME_MACHINE_TRACKER.registerAs("time_machine_tracker")
        TIME_MACHINE_RECALLER.registerAs("time_machine_recaller")
    }

    private fun Item.registerAs(path: String) = Registry.register(Registry.ITEM, identifier(path), this)
    private fun Block.createBlockItem() = BlockItem(this, FabricItemSettings().group(Mod.ITEM_GROUP))
}

